cookbooks/ascii_perturbation/documentation/tutorial_ascii.md

```diff
@@ -0,0 +1,329 @@
+Inside the Inner Radius (r<r_inner): The function calculates a value by adding the constant tcmb to a sinusoidal perturbation term. This perturbation term depends on the radial position r, the angular position theta, and the constants p and frequency.
+Formula:
+
+```{math}
+t_\text{cmb} + \left( p \cdot \sin \left( \frac{\pi \cdot (r - r_\text{inner})}{r_\text{outer} - r_\text{inner}} \right) \cdot \sin \left( \text{frequency} \cdot 2\pi \cdot \theta \right) \right)   ,
+
+```
+
+Outside the Outer Radius (r>r_outer): Like the inner radius case, but here the constant tsurf is added to the sinusoidal perturbation term.
+Formula:
+
+```{math}
+t_\text{surf} + \left( p \cdot \sin \left( \frac{\pi \cdot (r - r_\text{inner})}{r_\text{outer} - r_\text{inner}} \right) \cdot \sin \left( \text{frequency} \cdot 2\pi \cdot \theta \right) \right)  ,
+```
+
+Otherwise (For r Values Between r_inner and r_outer): Only the sinusoidal perturbation term is calculated.
+Formula:
+
+```{math} 
+p \cdot \sin \left( \frac{\pi \cdot (r - r_\text{inner})}{r_\text{outer} - r_\text{inner}} \right) \cdot \sin \left( \text{frequency} \cdot 2\pi \cdot \theta \right)  ,
+```
+
+This is the result for a quarter of shell:
+
+```{figure-md} fig:temperature
+<img src="temperature.png" style="width:48.0%" />
+
+Temperature for the above muparser function.
+```
+
+However, there may be a need to implement more complex temperature functions in our simulation that are difficult to reproduce with muparser language but can be easily crafted using the Python programming language. In such cases, it would be convenient to implement our perturbation using an ASCII file rather than a muparser function.
+For instance, we can modify the previous function to include numerous small perturbations along with three large perturbations (two cool and one warm, with a broader perturbation value), to induce a mega convection within our planet, while leaving the remainder undisturbed.
+Therefore, we may use the following mathematical function:
+
+```{math}
+temp = p_\text{large} \cdot \sin \left( \frac{\pi \cdot (r - r_\text{inner})}{r_\text{outer} - r_\text{inner}} \right) \cdot \sin \left( \text{frequency}_\text{large} \cdot \theta_\text{radians} \right) ,
+```
+
+And
+
+
+```{math}
+temp = p_\text{small} \cdot \sin \left( \frac{\pi \cdot (r - r_\text{inner})}{r_\text{outer} - r_\text{inner}} \right) \cdot \sin \left( \text{frequency}_\text{small} \cdot \theta_\text{radians} \right) ,
+```
+
+In the model, large temperature perturbations are designated to specific angular sections of a planetary ring to simulate intense convective movements. Conversely, the remaining sections are subjected to smaller perturbations.
+The final result will be as following:
+
+```
+<img src="ascii_temperature.png" style="width:48.0%" />
+
+Temperature distribution in ASCII format.
+```
+
+
+Now, let's write a Python script to utilize the function and export it in ASCII format
```

Comment:
Thank you for including the python code! However, as you have already provided it as a separate .py file in the cookbook folder, I think just a brief overview of what the script does would be fine in the cookbook folder.